processing/src/main/java/io/druid/query/aggregation/post/ExpressionPostAggregator.java

```diff
@@ -150,31 +148,28 @@ public String toString()
         .build();
   }
 
-  public static enum Ordering implements Comparator<Number>
+  public enum Ordering implements Comparator<Comparable>
   {
     // ensures the following order: numeric > NaN > Infinite
     numericFirst {
```

Comment:
The ordering is null-proof. If either lhs or rhs is null then the final else branch will be taken, which uses `Comparators.naturalNullsFirst()`.

I don't really have strong feelings about how the constructor should be implemented, but I don't want to change it in this PR, since it's not relevant to the main purpose. NB: If you (or anyone else) ends up reading this and doing another patch to change it, a very similar construction is used in ArithmeticPostAggregator.

I added a comment though explaining how the constructor works.